rasa_core_sdk/forms.py

```diff
@@ -435,6 +446,24 @@ def _activate_if_required(self, tracker):
             logger.debug("Activated the form '{}'".format(self.name()))
             return [Form(self.name())]
 
+    def _validate_prefilled_slots(self, dispatcher, tracker, domain):
+        # type: (Tracker) -> List[Dict]
+        """Upon form activation, validate all required slots that are already
+           filled."""
```

Comment:
should be blank line after method description. Also could you create one-liner description and the rest should go after blank line. `"""` in the end on the new line for multi-line comment, e.g.:
```
    @staticmethod
    def required_slots(tracker):
        # type: (Tracker) -> List[Text]
        """A list of required slots that the form has to fill.

        Use `tracker` to request different list of slots
        depending on the state of the dialogue
        """

        raise NotImplementedError(
            "A form must implement required slots " "that it has to fill"
        )
```

Reply:
Looks like this method duplicates `validate`, can we simply call `validate` upon activation, e.g. `_activate_if_required(...)` method

Reply:
I see, `validate` contains extraction. Then vise versa. let's call this from inside `validate`, renaming this method and passing slot names to be validated

Reply:
what do you think about introducing breaking changes and create separate `extract` method outside `validate`?

Reply:
if we call it from inside validate, it will re-validate all the slots it's already validated each time, no? I think it makes sense to keep this within the activation logic